OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1397,6 +1475,20 @@ internal static void UpgradeWeaponRules(ModData modData, int engineVersion, ref
 					if (node.Key == "BurstDelay")
 						node.Key = "BurstDelays";
 
+				// Multiply all health and damage in the RA mod by 100 to avoid issues caused by rounding
+				if (engineVersion < 20171125)
+				{
+					var mod = modData.Manifest.Id;
+					if (mod == "ra")
+					{
+						if (node.Key == "Damage" && parent.Value.Value == "SpreadDamage")
```

Comment:
`Damage` is (currently) an abstract class so it can be ignored, but you're right about the other two.

Reply:
I guess HPDW shouldn't be considered in this regard.

Reply:
Ah right, I forgot that it uses the "Damage" variable for the percentage and doesn't define it's own field. Looks like we only need to care about TargetDamage then.

Reply:
Hmm, I couldn't find an example for the `TargetDamage` and there is no `public readonly` in that file. Could you help me, because I don't know which name I should use for the `node.Key` in the upgrade rules.

Reply:
It uses the same `public readonly Damage` that is located on the abstract (i.e. not usable on its own) `DamageWarhead` that all other `*DamageWarhead`s inherit.

Just change the 'if' to
`if (node.Key == "Damage" && (parent.Value.Value == "SpreadDamage" || parent.Value.Value == "TargetDamage"))`

Reply:
Aha, thanks!